trie-db/src/triedbmut.rs

```diff
@@ -873,7 +1025,7 @@ where
 					// make an extension using it. this is a replacement.
 					InsertAction::Replace(Node::Extension(
 						existing_key.to_stored_range(common),
-						self.storage.alloc(Stored::New(augmented_low)).into()
+						self.storage.alloc(Stored::New(augmented_low)).into(),
 					))
 				}
 			},
```

Comment:
'fix' is checking a node is well formed, so I expect it can.
Removing the value of a branch is potentially making your branch 'one child no value' and replacing it by a leaf node. In this case if we run with eth variant, the extension before the node will need to be remove. I am not sure if the actual change is in fix or in fix sometime and in the remove visitor some other time.
But the extension case should only be touched when `TrieLayout` trait has associated constant `USE_EXTENSION` to true (which only happen with parity ethereum layout).